### ISSUE TITLE
How to turn off protection against concurrent migrations

### ISSUE THREAD

**Issue by JonPSmith** [dotnet/efcore#35186]
You have added code to throw an exception if multiple [migration executions happening simultaneously](https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-9.0/whatsnew#protection-against-concurrent-migrations). For my [AuthPermissions.AspNetCore](https://github.com/JonPSmith/AuthPermissions.AspNetCore) library (shorted to AuthP) I created a library to migrate multiples databases sequentially - see [RunStartupMethodsSequentially](https://github.com/JonPSmith/RunStartupMethodsSequentially). Unfortunately I can't find an way to turn off your concurrent migrations check.

What I, and the users who my library, I need a way to turn off the "concurrent migrations" check.

I know that you all think that applying migrations at deployment the the best approach, but I think it shouldn't be the ONLY way to migrate databases. I'm always trying to make it easier for developers to create their app, but I always provide a solution or say there are limitations. I have provided a robust way to deploy on startup in my [RunStartupMethodsSequentially](https://github.com/JonPSmith/RunStartupMethodsSequentially) library, and within my AuthP library have added a ["Down for maintenance" feature](https://github.com/JonPSmith/AuthPermissions.AspNetCore/wiki/%22Down-for-maintenance%22-feature) to allow the databases/setups can be updated while the application is live. These are both valid approaches.

**Comment by AndriySvyryd** [comment]
We don't support turning off migration locking. In most cases it should just make the operations run serially and it shouldn't interfere with operations that are already executed sequentially. So, if you are seeing exceptions, can you create a small repro so we can investigate whether this is a bug?

**Comment by JonPSmith** [comment]
Thanks @AndriySvyryd, 

I has a number of exceptions in my tests and example and I read that "We recommend applying migrations at deployment, rather than as <ins>part of application startup</ins>" and thought the errors I had was because .NET 9 version didn't allow migrating on startup. But your comment saying that "running migrations serially and it shouldn't interfere with operations" told me that something else it the problem.

NOTE: I apply the migrations using ASP.NET Core App startup feature. 

Once I have got things going I found the errors were from AuthP appling two migrations (serially) on some of my databases, e.g. the basic multi-tenant has one database with has two migrations: one migration to hold AuthP data and another migration to hold the tenant data. I needed to add the following code shown to some `DBContext` classes and it worked!

```c#
protected override void OnConfiguring( DbContextOptionsBuilder optionsBuilder)
{
    //This allows you to add more that one migration on this database
    optionsBuilder.ConfigureWarnings(x => x.Ignore(RelationalEventId.PendingModelChangesWarning));
    base.OnConfiguring(optionsBuilder);
}
```

Thanks for you help, and I'm very happy that the .NET 9 version of AuthP library only needs the developers to add the code above in a few places.
